app/Helpers/database/leaderboard.php

```diff
@@ -2,134 +2,165 @@
+            // No change made.
+            $retVal['BestScore'] = $existingLeaderboardEntry->Score;
         }
     } else {
-        $retVal['Success'] = false;
-        $retVal['Error'] = "Cannot find the leaderboard with ID: $lbID";
+        // No existing leaderboard entry. Let's insert a new one.
+        LeaderboardEntry::create([
+            'leaderboard_id' => $leaderboard->id,
+            'user_id' => $user->id,
+            'score' => $newEntry,
+        ]);
+
+        $retVal['BestScore'] = $newEntry;
     }
 
+    $retVal['TopEntries'] = GetLeaderboardEntriesDataJSON($lbID, $user->User, 10, 0, false);
+    $retVal['TopEntriesFriends'] = GetLeaderboardEntriesDataJSON($lbID, $user->User, 10, 0, true);
+    $retVal['RankInfo'] = GetLeaderboardRankingJSON($user->User, $lbID, (bool) $leaderboard->LowerIsBetter);
+
     return $retVal;
 }
 
-function removeLeaderboardEntry(string $user, int $lbID, ?int &$score): bool
-{
-    sanitize_sql_inputs($user);
-
-    $query = "SELECT le.Score, ld.Format FROM LeaderboardEntry AS le
-              LEFT JOIN LeaderboardDef AS ld ON ld.ID = le.LeaderboardID
-              LEFT JOIN UserAccounts AS ua ON ua.ID = le.UserID
-              WHERE ua.User = '$user' AND ld.ID = $lbID ";
-
-    $dbResult = s_mysql_query($query);
-    $data = mysqli_fetch_assoc($dbResult);
-
-    if (!$data) {
-        return false;
+/**
+ * @deprecated
+ * This is intended only for double writes.
+ * Delete when LeaderboardEntry is dropped in favor of leaderboard_entries.
+ */
+function writeLegacyLeaderboardEntry(
+    User $user,
+    Leaderboard $leaderboard,
+    int $newEntry,
+): void {
+    $existingLegacyLeaderboardEntry = LeaderboardEntryLegacy::where('LeaderboardID', $leaderboard->id)
+        ->where('UserID', $user->id)
+        ->first();
+
+    if ($existingLegacyLeaderboardEntry) {
+        // If the user is submitting a better score than their current entry,
+        // we'll override the current entry with their new score.
+        $comparisonOp = $leaderboard->LowerIsBetter === 1 ? '<' : '>';
+        $hasBetterScore =
+            ($comparisonOp === '<' && $newEntry < $existingLegacyLeaderboardEntry->Score)
+            || ($comparisonOp === '>' && $newEntry > $existingLegacyLeaderboardEntry->Score)
+        ;
+
+        if ($hasBetterScore) {
+            // Update the player's entry.
+            // This has to use `update()` because the table has a composite primary key.
+            LeaderboardEntryLegacy::where('LeaderboardID', $leaderboard->id)
+                ->where('UserID', $user->id)
+                ->update(['Score' => $newEntry, 'DateSubmitted' => now()]);
+        }
+    } else {
+        // No existing leaderboard entry. Let's insert a new one.
+        LeaderboardEntryLegacy::create([
+            'LeaderboardID' => $leaderboard->id,
+            'UserID' => $user->id,
+            'Score' => $newEntry,
+            'DateSubmitted' => now(),
+        ]);
```

Comment:
```
$ curl "localhost:64000/dorequest.php?r=submitlbentry&u=Jamiras&t=[redacted]&i=74041&s=1444&m=8e3630186e35d477231bf8fd50e54cdd&v=60253d7cbe23a30f406b0be85e1ddf7e"
{"Success":false,"Error":"Add [LeaderboardID] to fillable property to allow mass assignment on [App\\Models\\LeaderboardEntryLegacy].","Status":500}
```

Reply:
Looks like I accidentally didn't commit a model update which included that `$fillable` list. Resolved in latest.